export/export_pipeline/compute_collision_representation.py

```diff
@@ -0,0 +1,427 @@
+        with open(osp.join('export_pipeline', 'params.yaml'), 'r') as f:
+            self.p = yaml.safe_load(f)
+        self.input_dir = osp.join(scene_path, self.p['common']['GLTF_SCENE_DIR_NAME'])
+        self.output_dir = osp.join(scene_path, self.p['common']['COLLISION_DIR_NAME'])
+        for filename in next(os.walk(self.input_dir))[2]:
+            if not filename.endswith('.gltf'):
+                continue
+            if include_objects and not any([osp.splitext(filename)[0] == i for i in include_objects]):
+                continue
+            self.input_gltf_filenames.append(osp.join(self.input_dir, filename))
+
+
+    def run(self):
+        if not self.rerun:
+            shutil.rmtree(self.output_dir, ignore_errors=True)
+
+        bpy.ops.object.select_all(action='SELECT')
+        bpy.ops.object.delete()
+        
+        with open(osp.join(self.input_dir, 'actors_information.json'), 'r') as f:
+            actors_information = json.load(f)
+        
+        process_mesh_args = []
+        for input_gltf_filename in self.input_gltf_filenames:
+            bpy.ops.import_scene.gltf(filepath=input_gltf_filename)
+            obj = bpy.context.object
+            if obj.parent is not None:
+                continue
+            
+            object_name = obj.name.split('.')[0]
+            obj_dir = os.path.join(self.output_dir, object_name)
+            obj_info = actors_information[object_name]
+
+            joint_filename = input_gltf_filename.replace('.gltf', '_joints.json')
+            try:
+                with open(joint_filename, 'r') as f:
+                    joints_info = json.load(f)
+                joints_info = {j['name']: j for j in joints_info}
+            except FileNotFoundError:
+                joints_info = {}
+            
+            # traverse the tree, collect components info
+            geoms = []
+            components_info = {'static': {}, 'joints': {}}
+            # each element is (bpy object, directory_path, scale_factor, group_id)
+            # we apply the scale factor to the transforms here and save the resulting transforms in components.json.
+            # geometry scaling is already applied by UE when exporting geometries to GLTF.
+            q = deque([(obj, obj_dir, np.ones(3), object_name), ])
+            while len(q):
+                o, dir, scale_factor, group_id = q.popleft()
+                name = o.name.split('.')[0]
+                xyz, quat = get_relative_pose_and_center(o, scale_factor)
+
+                if o.children or ((o.type == 'MESH') and (o.data.name != 'SM_Dummy')):
+                    if name in components_info['static']:
+                        raise AssertionError(f'repeated static mesh {name}')
+                    components_info['static'][name] = {'pos': xyz, 'quat': quat}
+                    os.makedirs(dir, exist_ok=True)
+                    if o.children:  # StaticMeshComponent without any geometry (meant for grouping)
```

Comment:
A `StaticMeshComponent` can have non-trivial geometry and also have children. Is this comment up-to-date?